taxonomy/management/commands/refresh_xblock_skills.py

```diff
@@ -91,6 +100,32 @@ def is_valid_key(key, key_cls, key_cls_str):
             LOGGER.error('[TAXONOMY] Invalid %s: [%s]', key_cls_str, key)
         return False
 
+    @staticmethod
+    def is_course_already_processed(course_run_key: str):
+        return CourseRunXBlockSkillsTracker.objects.filter(course_run_key=course_run_key).exists()
+
+    @staticmethod
+    def mark_course_completed(course_run_key: str, success_count: int, failure_count: int, threshold: float) -> bool:
+        """
+        Add entry in this table marking the course complete if the ratio of
+        success_count/total > threshold
+        """
+        total = success_count + failure_count
+        success_ratio = success_count / total if total else 0
+        if success_ratio < threshold:
+            return False
+        LOGGER.info(
+            '[TAXONOMY] Marking course run: [%s] as complete as success ratio: [%s] is greater than threshold: [%s]',
+            course_run_key,
+            success_ratio,
+            threshold
+        )
+        CourseRunXBlockSkillsTracker.objects.update_or_create(
+            course_run_key=course_run_key,
+            defaults={"course_run_key": course_run_key}
+        )
```

Comment:
Since we are not storing anything in the table other than the course key itself
```suggestion
        CourseRunXBlockSkillsTracker.objects.get_or_create(course_run_key=course_run_key)
```

Reply:
Updated.